ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -384,10 +384,18 @@ private BlockHeader getOmmer(final BlockBody blockBody, final int index) {
     return blockchain.getBlockHashByNumber(blockNumber).flatMap(this::blockByHashWithTxHashes);
   }
 
+  public Optional<BlockHeader> getBlockHeaderByHash(final Hash hash) {
```

Comment:
Why did you add the ByHash? I think that is obvious, because the only argument to the method is the block hash ...